### ISSUE TITLE
Non-working Constant Propagation for Subregisters

### ISSUE THREAD

**Issue by mrphrazer** [Vector35/binaryninja-api#2766]
**Binary Ninja Version**
Version 2.5.3112-dev (Build ID dd1a9553)

**Describe the bug**
The constant propagation does not propagate byte constants if the main registers are not directly set (like `r8` set to `0x0`).
```
00000000  41b0a9             mov     r8b, 0xa9
00000003  4180e0a0           and     r8b, 0xa0
00000007  4180e8e2           sub     r8b, 0xe2
0000000b  4180c0b6           add     r8b, 0xb6
0000000f  41f6d0             not     r8b
00000012  490fb6c0           movzx   rax, r8b
00000016  c3                 retn     {__return_addr}
```
The LLIL in SSA form already knows that everything relies on `r8b`:

```
   0 @ 00000000  r8#1.r8b = 0xa9
   1 @ 00000003  r8#2.r8b = r8#1.r8b & 0xa0
   2 @ 00000007  r8#3.r8b = r8#2.r8b - -0x1e
   3 @ 0000000b  r8#4.r8b = r8#3.r8b - 0x4a
   4 @ 0000000f  r8#5.r8b = not.b(r8#4.r8b)
   5 @ 00000012  rax#1 = zx.q(r8#5.r8b)

```

At latest the MLILform only operatoes on the byte-level and should be able to propagate it.
```
   0 @ 00000000  char r8 = 0xa9
   1 @ 00000003  char r8 = r8 & 0xa0
   2 @ 00000007  char r8 = r8 - 0xe2
   3 @ 0000000b  char r8 = r8 - 0x4a
   4 @ 0000000f  char r8 = not.b(r8)
   5 @ 00000012  uint64_t rax = zx.q(r8)
```
**Expected behavior**

Propagte the constant and simplify the result to  `0x8b`.

**Questions**

* On which level is the constant propagation ususally applied?
* Is it possible to individually enforce the constant propagation on other IL levels (like 'do that on MLIL for function X')?

**Comment by bpotchik** [comment]
This actually works as designed at the moment and the PossibleValueSet dataflow system is used to track and propagate the value.

FWIW, this is also the same issue as https://github.com/Vector35/binaryninja-api/issues/921 and I have not had a chance to look back into this yet.

**Comment by mrphrazer** [comment]
Thanks for the fast reply.

If it works on the PossibleValueSet, it should be possible to propagate the first char assignment downward on MLIL level? With user-informed dataflows on the MLIL level, this always worked. But even if I additionally set a user variable variable for `   0 @ 00000000  char r8 = 0xa9`, it still doesn't propagate.

**Comment by bpotchik** [comment]
Fixed in `3.2.3934-dev`.

I did not mark constant propagated partial reg expressions for elimination yet, but will probably push that shortly.

**Comment by ccarpenter04** [comment]
I'm quite happy to see this resolved, thank you for taking the time to work on it :)
